### DISCUSSION TITLE
MQTT Collector - 0 Values

### DISCUSSION THREAD

**Discussion by loebse** [solectrus/solectrus#2183] [Q&A]
I remeber from this discussion https://github.com/orgs/solectrus/discussions/1620 that there was an issue when using the MQTT collector and receiving 0-values for one or the other data point.

Reading through this discussion it is mentioned that it was implemented optionally. However, I think this was before the refactoring by @ledermann, correct?

I can see some inconsistencies for 0-values in the current version (0.12.0). Is this still optional and needs some special config?

**Comment by ledermann** [comment]
I apologize for the delayed response as I was on vacation :-)

Regarding the 0-values: this concerns the measurements for the grid (feed-in vs. consumption) and the battery (charging vs. discharging). Only one value can be given at any time, the other is automatically set to 0. The code for this is still present and can be found here:

Battery: https://github.com/solectrus/mqtt-collector/blob/develop/app/mapper.rb#L64-L75
Grid: https://github.com/solectrus/mqtt-collector/blob/develop/app/mapper.rb#L77-L87

This is part of the MQTT-Collector. The current version 0.12.0 of the dashboard should not show any different behavior than older versions. What specific problems are occurring?

**Reply by loebse** [reply]
Now I had been off for some time ;)

I am facing the behaviour that the SOC of the battery show 0% from time to time and then a few mins later jumps back to the actual value:

<img width="592" alt="image" src="https://github.com/solectrus/solectrus/assets/39953358/ec288b21-f64f-4195-aca7-1c9267483ad2">

As far as I remember, this was an issue when the MQTT collector does not get an updated value for some time for this data point. As the ioBroker plugin reads some of the values (e.g. SOC) only every 5 or 10 mins, this might lead to this behaviour.

**Reply by ledermann** [reply]
Thanks for bringing this to my attention. I think this is caused by the InfluxDB query used by SOLECTRUS for getting the **latest** measurements. It uses a range starting `5 minutes ago`, so older data will be excluded:
https://github.com/solectrus/solectrus/blob/v0.12.0/app/services/power_sum.rb#L6

I have created a separate issue for this (see #2238) and will fix it soon.
